share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -198,7 +201,17 @@ function(ExternalZephyrProject_Add)
     # Only set image specific board if provided.
     # The sysbuild BOARD is exported through sysbuild cache, and will be used
     # unless <image>_BOARD is defined.
-    list(APPEND sysbuild_cache_strings "${ZBUILD_APPLICATION}_BOARD:STRING=${ZBUILD_BOARD}\n")
+    if(DEFINED ZBUILD_BOARD_REVISION)
```

Comment:
I think ZBUILD_BOARD it should be carried with it`s board reversion, no need add a new parameter named ZBUILD_BOARD_REVISION.

that means if user define the board, use user define board, if user not passing board parameter using the current board, if current board have an reversion, add board and reversion, or only current board used.

Reply:
This is for boards that have multiple cores, i.e. 5340, so that the other core can be used but the overall revision of the same board can still be passed down

Reply:
I'm aligned with @Cherish-Gww on this.

Mainly because it's generally bad practice to rely on global settings inside functions, a function should be self contained so that you know what it will be using just from API.

Also imagine a future use-case where someone wants to use sysbuild for multiple images for independent devices, as mentioned here, bullet 4: https://github.com/zephyrproject-rtos/zephyr/pull/40555#issuecomment-1165476795
(you most likely need to `load more ...` in github before link takes you to correct comment)

then independent devices could use different revisions, so for future proof, passing board revision as part of function call will be preferred.

Reply:
Fixed